File: app/src/main/java/com/lazycouple/restapiclient/ui/module/ContextModule.java
package com.lazycouple.restapiclient.ui.module;

import android.content.Context;

import dagger.Module;
import dagger.Provides;

/**
 * Created by amuyu on 2017. 6. 21..
 */
@Module
public class ContextModule {

    private final Context context;

    public ContextModule(Context context) {
        this.context = context;
    }

    @Provides
    public Context provideContext() {
        return this.context;
    }
}

File: app/src/main/kotlin/com/lazycouple/restapiclient/ui/module/ContextModule.kt
package com.lazycouple.restapiclient.ui.module

import android.content.Context

import dagger.Module
import dagger.Provides

/**
 * Created by amuyu on 2017. 6. 21..
 */
@Module
class ContextModule(private val context: Context) {

    @Provides
    fun provideContext(): Context {
        return this.context
    }
}
